sshd-core/src/main/java/org/apache/sshd/common/channel/ChannelOutputStream.java

```diff
@@ -171,43 +227,52 @@ public synchronized void write(byte[] buf, int s, int l) throws IOException {
                                 "Interrupted while waiting for remote space on write len=" + l + " to " + this)
                                         .initCause(e);
                     }
-                }
-                session.resetIdleTimeout();
-                continue;
+                    session.resetIdleTimeout();
+                    break;
+                default:
+                    // BUFFERED implies l == 0; outer loop will terminate
+                    break;
```

Comment:
No need for *break* in *default* clause.

Reply:
Not done. I prefer to have *every* switch branch be terminated by a break, return, or throw. (I can't remember the last time I had the urge to use a fall-though.) It may not be necessary here, but it doesn't hurt.